references/classification/quantization/train.py

```diff
@@ -0,0 +1,255 @@
+                                                             valdir,
+                                                             args.batch_size,
+                                                             args.eval_batch_size,
+                                                             args.workers,
+                                                             args.cache_dataset,
+                                                             args.distributed)
+
+    print("Creating model", args.model)
+    model = torchvision.models.quantization.__dict__[args.model](pretrained_float_model=True)
+    if args.test_only is False:
+        model.fuse_model()
+        model.qconfig = torch.quantization.get_default_qat_qconfig(args.backend)
+        torch.quantization.prepare_qat(model, inplace=True)
+        model.to(device)
+        if args.distributed and args.sync_bn:
+            model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+
+    criterion = nn.CrossEntropyLoss()
+
+    optimizer = torch.optim.SGD(
+        model.parameters(), lr=args.lr, momentum=args.momentum,
+        weight_decay=args.weight_decay)
+    if args.apex:
+        model, optimizer = amp.initialize(model, optimizer,
+                                          opt_level=args.apex_opt_level
+                                          )
+
+    lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer,
+                                                   step_size=args.lr_step_size,
+                                                   gamma=args.lr_gamma)
+
+    model_without_ddp = model
+    if args.distributed:
+        model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.gpu])
+        model_without_ddp = model.module
+
+    if args.resume:
+        checkpoint = torch.load(args.resume, map_location='cpu')
+        model_without_ddp.load_state_dict(checkpoint['model'])
+        optimizer.load_state_dict(checkpoint['optimizer'])
+        lr_scheduler.load_state_dict(checkpoint['lr_scheduler'])
+        args.start_epoch = checkpoint['epoch'] + 1
+
+    if args.test_only:
+        model.to(device)
+        model.eval()
+        model.fuse_model()
+        model.qconfig = torch.quantization.get_default_qconfig(args.backend)
+        torch.quantization.prepare(model, inplace=True)
+        # Calibrate first
+        evaluate(model, criterion, data_loader, device=device,
+                 neval_batches=args.num_calibration_batches)
+        torch.quantization.convert(model, inplace=True)
+        if args.output_dir:
+            if utils.is_main_process():
+                torch.save(model.state_dict(), os.path.join(args.output_dir,
+                           'quantized_post_train_model.pth'))
+        print('Saving quantized model')
+        evaluate(model, criterion, data_loader_test, device=device)
+        return
+
+    model.train()
+    model.apply(torch.quantization.enable_observer)
+    model.apply(torch.quantization.enable_fake_quant)
```

Comment:
Will the training code for torchscript quantization be the same, or are those changes specific to eager quantization?

Reply:
I would expect this part to be the same. torchscript quantization support for quantization aware training is still work in progress